flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/source/row/converter/PostgresRowConverter.java

```diff
@@ -28,4 +34,39 @@
 	public PostgresRowConverter(RowType rowType) {
 		super(rowType);
 	}
+
+	@Override
+	public JDBCFieldConverter createConverter(LogicalType type) {
+		LogicalTypeRoot root = type.getTypeRoot();
+
+		if (root == LogicalTypeRoot.ARRAY) {
+			ArrayType arrayType = (ArrayType) type;
+			LogicalTypeRoot elemType = arrayType.getElementType().getTypeRoot();
+
+			if (elemType == LogicalTypeRoot.VARBINARY) {
```

Comment:
I would suggest to use `LogicalTypeChecks#hasFamily(elemType, LogicalTypeFamily.BINARY_STRING)` to also support `BINARY`.